flink-ml-core/src/test/java/org/apache/flink/ml/common/window/WindowsTest.java

```diff
@@ -0,0 +1,187 @@
+import org.apache.flink.streaming.api.windowing.windows.Window;
+import org.apache.flink.test.util.AbstractTestBase;
+import org.apache.flink.util.Collector;
+
+import org.apache.commons.collections.IteratorUtils;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/** Tests the {@link Windows}s and {@link WindowUtils}. */
+@SuppressWarnings("unchecked")
+public class WindowsTest extends AbstractTestBase {
+    private static final int RECORD_NUM = 100;
+
+    private static List<Long> inputData;
+
+    private static DataStream<Long> inputStream;
+    private static DataStream<Long> inputStreamWithInterval;
+    private static DataStream<Long> inputStreamWithTimestamp;
+
+    @BeforeClass
+    public static void before() {
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        inputData = new ArrayList<>();
+        for (long i = 0; i < RECORD_NUM; i++) {
+            inputData.add(i);
+        }
+        inputStream = env.fromCollection(inputData);
+
+        inputStreamWithInterval =
+                inputStream
+                        .map(
+                                new MapFunction<Long, Long>() {
+                                    private int count = 0;
+
+                                    @Override
+                                    public Long map(Long value) throws Exception {
+                                        count++;
+                                        if (count % (RECORD_NUM / 2) == 0) {
+                                            Thread.sleep(1000);
+                                        }
+                                        return value;
+                                    }
+                                })
+                        .setParallelism(1);
+
+        inputStreamWithTimestamp =
+                inputStream.assignTimestampsAndWatermarks(
+                        WatermarkStrategy.<Long>forMonotonousTimestamps()
+                                .withTimestampAssigner(
+                                        (SerializableTimestampAssigner<Long>)
+                                                (element, recordTimestamp) -> element));
+    }
+
+    @Test
+    public void testBoundedWindow() throws Exception {
+        DataStream<List<Long>> outputStream =
+                WindowUtils.windowAllProcess(
+                        inputStream, null, new CreateAllWindowBatchFunction<>());
+        List<List<Long>> actualBatches = IteratorUtils.toList(outputStream.executeAndCollect());
+        assertEquals(1, actualBatches.size());
+        assertEquals(new HashSet<>(inputData), new HashSet<>(actualBatches.get(0)));
+    }
+
+    @Test
+    public void testCountWindow() throws Exception {
```

Comment:
Would it be better to name the test based on the windows subclass name?

Same for other tests.